stories/automilos.fmf

```diff
@@ -0,0 +1,15 @@
+story:
+    As AutoMilos I want to execute relevant tests when the issue is switched to given state
+
+description:
+    Test/plan run on AutoMilos should be triggered when the issue is switched to given state.
+    The trigger should be done via the trigger-url provided in the JIRA issue. There could be the standard url for the
+    service used by AutoMilos with the additional query parameters specific to AutoMilos needs. The url should have all
+    the needed info to run the test, plan, or combination of both together in Testing farm.
```

Comment:
I don't think this story is really necessary for AutoMiloš (and seems to descriibe something different from what is documented in TT-261).  AutoMiloš currently works as a scheduled job that regularly goes through the relevant issues and decides what needs to be done based on the state of each issue. This works sufficiently well for the current needs and I wouldn't want to switch to a webhook kind of trigger (it would require a significant overhaul).

It really just needs a defined way to identify relevant tests for a given issue and that seems to be already well covered in `stories/link.fmf`.

Reply:
Thanks for clarification. So, I guess, the first part of the description could be dropped. Perhaps, it would make sense to clearly define here which `testing-farm` options is `AutoMilos` using to schedule relevant tests? Is it using `--plan`, or `--plan-filter`?  Or both? And also `--test-filter` I guess? And definitely `--git-url` and `--git-ref`. Anything more?

Reply:
I think that AutoMilos's primary focus is on "trivial" reproducers where executing a single test case is sufficient. If I should project keylime test practices on AutoMilos we would need to:

1. choose a specific plan
2. filter out specific (multiple) test cases from this plan and execute them

From this perspective both `--plan-filter` and `--test-filter` would be needed. But TBH, I am still not convinced that maintaining this kind of data in Jira is the best approach as we might be too much focused on "doing things the way we did in the past".

Reply:
Unfortunately, AutoMiloš hasn't been ported to TFT yet (it still uses the legacy TCMS + workflow-tomorrow), and I haven't yet done much experimenting/scoping on the TFT support, so I'm not sure which options would be needed in the end. (BTW, the last time I looked at it, I thought it might be better to use the REST API directly - from Python it should be quite easy and possibly less error-prone than programmatically interacting with the CLI. So please consider the possibility that it would use the API instead of the CLI, if that's relevant.)

Reply:
Ok, as discussed on the sync meeting, for now let's just stick with the simple case of selecting test and plan by name. This should work basically in the same way both for API and CLI. If needed in the future, we can extend with `plan-filter` and `test-filter` as well. Probably does not make sense to dive into this more as we will need to clarify how much AutoMilos will be obsoleted by the pull request workflow in GitLab.